plasmapy/particles/particle_class.py

```diff
@@ -111,13 +111,31 @@ def mass(self) -> Union[u.Quantity, Real]:
     def charge(self) -> Union[u.Quantity, Real]:
         raise NotImplementedError
 
+    @property
+    def particle_dict(self) -> dict:
+        particle_dictionary = {
+            "plasmapy_particle": {
+                "type": type(self).__name__,
+                "description": {
+                    # subclass specific description
+                }
+            }
+        }
```

Comment:
```suggestion
        particle_dictionary = {
            "plasmapy_particle": {
                "type": type(self).__name__,
                "module": self.__module__,
                "mass": self.mass.__str__(),
                "charge": self.charge.__str__(),
            }
        }
```

Since `"description"` is not universal among all particle classes and is not used in reconstructing the object, I would leave it out.   Every particle will have `mass` and `charge` so this should be here by default.  `module` is for origin tracing.  You could also add things similar to `created_date`.  Basically, this should only contain data that helps trace the source/generation of the json file/particle and data needed to reconstruct the particle.

Reply:
In the implementation of the deserialization, which I have, unfortunately, kept separately in PR #836 , I use the `description` key as the index for the initialization parameters.

The reason I haven't kept  `mass` or `charge` as default is because the Particle class throws a ChargeError when there is an attempt to retrieve its charge. I could make `mass` default, but reasoned that it may be easier for developers to understand the concept for the overriding of the `particle_dict` property. (i.e overriding the CustomParticle's `particle_dict` property to add `mass` and `charge` versus just `charge). I'd like to ask for advice regarding this bit.

Thanks for the suggestion regarding the date and `module`. I can include them in the JSON representation.

Reply:
Interesting...  Once you get everything into a single PR I'll take a look into where the error is coming from.